src/keboola_mcp_server/tools/components/model.py

```diff
@@ -1,354 +1,550 @@
+
+# ============================================================================
+# TYPE DEFINITIONS
+# ============================================================================
 
 ComponentType = Literal['application', 'extractor', 'writer']
 TransformationType = Literal['transformation']
 AllComponentTypes = Union[ComponentType, TransformationType]
 
 
-class ConfigToolOutput(BaseModel):
-    component_id: str = Field(description='The ID of the component.')
-    configuration_id: str = Field(description='The ID of the configuration.')
-    description: str = Field(description='The description of the configuration.')
-    timestamp: datetime = Field(description='The timestamp of the operation.')
-    # success is always true unless the tool fails - to inform agent and prevent need to fetch objects
-    success: bool = Field(default=True, description='Indicates if the operation succeeded.')
-    links: list[Link] = Field(description='The links relevant to the configuration.')
-
+# ============================================================================
+# COMPONENT MODELS
+# ============================================================================
 
-class ReducedComponent(BaseModel):
-    """
-    A Reduced Component containing basic information about the Keboola Component and its capabilities.
-    This model is used in list views or when only basic component information is needed.
+class ComponentCapabilities(BaseModel):
     """
+    Component capabilities derived from developer portal flags.
 
-    component_id: str = Field(
-        description='The ID of the component',
-        validation_alias=AliasChoices('id', 'component_id', 'componentId', 'component-id'),
-        serialization_alias='componentId',
-    )
-    component_name: str = Field(
-        description='The name of the component',
-        validation_alias=AliasChoices(
-            'name',
-            'component_name',
-            'componentName',
-            'component-name',
-        ),
-        serialization_alias='componentName',
-    )
-    component_type: str = Field(
-        description='The type of the component',
-        validation_alias=AliasChoices('type', 'component_type', 'componentType', 'component-type'),
-        serialization_alias='componentType',
-    )
-
-    component_flags: list[str] = Field(
-        default_factory=list,
-        description='List of developer portal flags.',
-        validation_alias=AliasChoices('flags', 'component_flags', 'componentFlags', 'component-flags'),
-        serialization_alias='componentFlags',
-    )
+    Represents what a component can do in terms of data processing:
+    - Row-based: Can have multiple configuration rows for different tasks
+    - Table I/O: Can read from or write to data tables
+    - File I/O: Can read from or write to files
+    - OAuth: Requires OAuth authentication setup
+    """
 
-    # Capability flags derived from component_flags
     is_row_based: bool = Field(
```

Comment:
If a field definition fits into a single line, I'd use a single line instead of 4.